server/src/main/java/org/apache/cassandra/sidecar/routes/cdc/StreamCdcSegmentHandler.java

```diff
@@ -0,0 +1,195 @@
+                                               HttpResponse response,
+                                               InstanceMetadata instance)
+    {
+        long segmentFileLength = segmentFile.length();
+        return getOrCreateLinkedCdcFilePairAsync(segmentFile, indexFile)
+               .compose(cdcFilePair ->
+                        openCdcIndexFileAsync(cdcFilePair)
+                        .compose(cdcIndex -> {
+                            // stream the segment file; depending on whether the cdc segment is complete or not, cap the range of file to stream
+                            String rangeHeader = context.request().getHeader(HttpHeaderNames.RANGE);
+                            return cdcIndex.isCompleted
+                                   ? fileStreamer.parseRangeHeader(rangeHeader, segmentFileLength)
+                                   : fileStreamer.parseRangeHeader(rangeHeader, cdcIndex.latestFlushPosition);
+                        })
+                        .compose(range -> fileStreamer.stream(response, instance.id(), cdcFilePair.segmentFile.getAbsolutePath(), segmentFileLength, range))
+               );
+    }
+
+    @Override
+    protected String extractParamsOrThrow(RoutingContext context)
+    {
+        return context.request().getParam("segment");
+    }
+
+    private void validateCdcSegmentFile(File segmentFile) throws HttpException
+    {
+        // We only accept stream request for log files
+        if (!isValid(segmentFile.getName()) || !isLogFile(segmentFile.getName()))
+        {
+            throw wrapHttpException(HttpResponseStatus.BAD_REQUEST, "Invalid path param for CDC segment: " + segmentFile.getName());
+        }
+
+        // check file existence
+        if (!segmentFile.exists())
+        {
+            throw wrapHttpException(HttpResponseStatus.NOT_FOUND, "CDC segment not found: " + segmentFile.getName());
+        }
+
+        // check file content
+        long fileSize = segmentFile.length();
+        if (fileSize == 0)
+        {
+            throw wrapHttpException(HttpResponseStatus.REQUESTED_RANGE_NOT_SATISFIABLE, "File is emtpy");
+        }
+    }
+
+    private Future<CdcFilePair> getOrCreateLinkedCdcFilePairAsync(File segmentFile, File indexFile)
+    {
+        return serviceExecutorPool.executeBlocking(() -> {
+            // hardlink the segment and its index file,
+            // in order to guarantee the files exist in the subsequent calls (within the cache duration)
+            try
+            {
+                File linkedSegmentFile = cdcLogCache.createLinkedFileInCache(segmentFile);
+                File linkedIndexFile = cdcLogCache.createLinkedFileInCache(indexFile);
+                return new CdcFilePair(linkedSegmentFile, linkedIndexFile);
+            }
+            catch (IOException e)
+            {
+                throw wrapHttpException(HttpResponseStatus.INTERNAL_SERVER_ERROR, "Failed to prepare CDC segment to stream", e);
```

Comment:
do we want to log the error here for debugging purposes if an issue is encountered?

Reply:
Added log.